lib/src/components/NxFileUpload/NxMultiFileUpload/NxMultiFileUpload.tsx

```diff
@@ -0,0 +1,180 @@
+
+    if (normalizedFiles) {
+      forEach((f) => dataTransferObject.items.add(f), Array.from(normalizedFiles));
+    }
+
+    onChangeProp(dataTransferObject.files);
+  }
+
+  function openPicker() {
+    inputRef.current?.click();
+  }
+
+  function onDismiss(fileObj: File) {
+    const closeBtns = Array.from(selectedFilesContainerRef.current?.
+        querySelectorAll<HTMLButtonElement>('.nx-selected-file__dismiss-btn') ?? []);
+
+    if (files) {
+      const filesArray = Array.from(files);
+      filesArray.map((f) => {
+        if (f === fileObj) {
+          const idx = filesArray.indexOf(f);
+          if (idx === files.length - 1) {
+            // if the selected file is the last in the container, set the focus to the next last dismiss button
+            // if selected file is only file, set focus on the input
+            closeBtns.length === 1 ? inputRef.current?.focus() : closeBtns[idx - 1].focus();
+          }
+          else {
+            // need to explicity set focus on next closeBtn for Safari
+            closeBtns[idx].focus();
+          }
+        }
+      });
+    }
+
+    const dataTransferObject = new DataTransfer();
+    if (inputRef.current) {
+      if (files) {
+        forEach((f: File) => dataTransferObject.items.add(f), without([fileObj], Array.from(files)));
+
+        const normalizedFiles = !dataTransferObject.files.length ? null : dataTransferObject.files;
+        onChangeProp(normalizedFiles);
+      }
+    }
+  }
+
+  useEffect(function() {
+    if (inputRef.current) {
+      if (isFileSelected) {
+        inputRef.current.files = files;
+      }
+      else {
+        // there's no way to clear the list via the files prop, but this does it
+        inputRef.current.value = '';
+      }
+    }
+  }, [files]);
+
+  return (
+    <div ref={ref} className={className} >
+      <div className="nx-multi-file-upload__container">
+        <input ref={inputRef}
+               { ...attrs }
+               disabled={disabled}
+               onChange={onChange}
+               id={inputId}
+               className="nx-multi-file-upload__input"
+               type="file"
+               aria-label={isFileSelected ? `${files.length} files selected` : 'No File Selected'}
+               aria-required={isRequired ?? undefined}
+               aria-invalid={showError || undefined}
+               aria-errormessage={showError ? validationErrorId : undefined}
+               multiple/>
+        {/* Keynav and screenreaders can ignore the button itself in favor of the <input> */}
+        <NxButton type="button"
+                  aria-hidden={true}
+                  disabled={disabled}
+                  variant="tertiary"
+                  onClick={openPicker}
+                  tabIndex={-1}
+                  className="nx-multi-file-upload__select-btn">
+          Add Files
+        </NxButton>
+        <div ref={selectedFilesContainerRef} className="nx-multi-file-upload__container__files nx-scrollable">
```

Comment:
This should probably be an `<ol>` and the selected files wrapped in `<li>`s.  It also seems like it would make sense to link the `<input>` to this list via `aria-controls`.

Reply:
> This should probably be an `<ol>` and the selected files wrapped in `<li>`s. It also seems like it would make sense to link the `<input>` to this list via `aria-controls`.

Implemented here- [37a645a](https://github.com/sonatype/sonatype-react-shared-components/pull/850/commits/37a645a642ce6190b53c38de6de5268a7993ad9a). I changed the wrapper `nx-file-upload__no-file-message` from a `span` to an `li` to prevent too much nesting.